.golangci.yaml

```
@@ -0,0 +1,39 @@
+run:
+  timeout: 30m
+  skip-files:
+    - "^zz_generated.*"
+
+issues:
+  max-same-issues: 0
+  # Excluding configuration per-path, per-linter, per-text and per-source
+  exclude-rules:
+    # exclude ineffassing linter for generated files for conversion
+    - path: conversion\.go
+      linters:
+        - ineffassign
+
+linters:
+  disable-all: true
+  enable: # please keep this alphabetized
+  # Don't use soon to deprecated[1] linters that lead to false
+  # https://github.com/golangci/golangci-lint/issues/1841
+  # - deadcode
+  # - structcheck
+  # - varcheck
+    - ineffassign
+    - staticcheck
+    - unused
+
+linters-settings: # please keep this alphabetized
+  staticcheck:
+    go: "1.17"
+    checks: [
+      "all",
+      "-S1*",    # TODO(fix) Omit code simplifications for now.
+      "-ST1*",   # Mostly stylistic, redundant w/ golint
+      "-SA5011", # TODO(fix) Possible nil pointer dereference
+      "-SA1019", # TODO(fix) Using a deprecated function, variable, constant or field
+      "-SA2002"  # TODO(fix) Called testing.T.FailNow or SkipNow in a goroutine, which isn’t allowed
```

```suggestion
      "-ST1000",  # Incorrect or missing package comment
      "-ST1003",  # Poorly chosen identifier
      "-ST1005",  # Incorrectly formatted error string
      "-ST1006",  # Poorly chosen receiver name
      "-ST1012",  # Poorly chosen name for error variable
      "-ST1016",  # Use consistent method receiver names
      "-ST1020",  # Comment on exported method should be of the form ...
      "-ST1021",  # Comment on exported type should be of the form ...
      "-ST1022",  # Comment on exported const should be of the form ...
      "-SA5011"   # TODO(fix) Possible nil pointer dereference
```

Pulling these from the existing hack/verify-staticcheck.sh:

https://github.com/kubernetes/kops/blob/88a9d50e7321b185c675540f7035dd203e450a44/hack/verify-staticcheck.sh#L30-L42